File: 958-sort-array-by-parity-ii/sort-array-by-parity-ii.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) 
    {
        int n=nums.size();
        int even=0;
        int odd=1;
        int wrongeven=-1;
        int wrongodd=-1;
        while(even<n && odd<n)
        {
            while(even<n && nums[even]%2==0)
                even=even+2;
            while(odd<n && nums[odd]%2!=0)
                odd=odd+2;
            if(even<n && odd<n)
            {
                swap(nums[even],nums[odd]);
            }
            
        }
        return nums;
    }
};

File: 958-sort-array-by-parity-ii/sort-array-by-parity-ii.py
class Solution(object):
    def sortArrayByParityII(self, a):
        i=0
        j=1
        while(i<len(a) and j<len(a)):
            while(i<len(a) and a[i]%2==0):
                i=i+2
            while(j<len(a) and a[j]%2!=0):
                j=j+2
            if(i<len(a) and j<len(a)):
               a[i],a[j]=a[j],a[i]

             
        return a
        